### ISSUE TITLE
IOS voiceover accessibility problem when mode is card

### ISSUE THREAD

**Issue by lefterisk** [react-navigation/react-navigation#9326]
**Current Behavior**

- When stack Navigator is in default (card) mode the first screen that loads (default) works as expected with voiceover, but all subsequent screens do not. The Voiceover cant focus any elements on the screen and gestures dont work. Even if one taps in the screen and manually focuses an element, voiceover cannot figure out if there are other elements on the screen and even ignores ally labels (accessible etc)

**Expected Behavior**

- All screens should behave similarly when used with voiceover 

**How to reproduce**

- Simply have a StackNavigator with multiple screens. The default/first screen will work as expected with IOS voiceover, but when navigating to other pages voiceover will no longer be able to traverse the screens properly.  

**Your Environment**

| software                       | version |
| ------------------------------ | ------- |
| iOS                 |   14.4
| @react-navigation/native       | 5.8.10
| @react-navigation/stack        | 5.13.0
| react-native-gesture-handler   | 1.9.0
| react-native-safe-area-context | 3.1.9
| react-native-screens           | 2.16.1
| react-native                   | 0.63.3
| node                           | 12.2
| yarn                    | 1.22.10

As this problem only occurs when mode is `card` I managed to trace the issue to this line of code https://github.com/react-navigation/react-navigation/blob/8ee0dda155b4cde43be1e58170e44823b54e7d4f/packages/stack/src/TransitionConfigs/CardStyleInterpolators.tsx#L58 

It seems that the animation logic for `SlideFromRightIOS` is somehow messing up with the focusability of the screen. If you comment this line out it all seems to work.

**Comment by lefterisk** [comment]
Upgraded to the latest and the issue persists

**Comment by lefterisk** [comment]
Further tinkering revealed that disabling `enableScreens()` from the app root also fixes the issue

**Comment by WoLewicki** [comment]
@lefterisk could you provide a minimal reproduction of the issue to help us fix it?

**Comment by satya164** [comment]
Closing due to no response
